pandas/core/window/rolling.py

```diff
@@ -544,19 +533,32 @@ def aggregate(self, func, *args, **kwargs):
     agg = aggregate
 
 
-class BaseWindowGroupby(GotItemMixin, BaseWindow):
+class BaseWindowGroupby(BaseWindow):
     """
     Provide the groupby windowing facilities.
     """
 
-    def __init__(self, obj, *args, **kwargs):
-        kwargs.pop("parent", None)
-        groupby = kwargs.pop("groupby", None)
-        if groupby is None:
-            groupby, obj = obj, obj._selected_obj
-        self._groupby = groupby
-        self._groupby.mutated = True
-        self._groupby.grouper.mutated = True
+    _attributes = [
+        "_grouping_indices",
+        "_grouping_keys",
+        "_grouping_codes",
+        "_grouping_levels",
+    ]
+
+    def __init__(
+        self,
+        obj,
+        *args,
+        _grouping_indices=None,
+        _grouping_keys=None,
+        _grouping_codes=None,
+        _grouping_levels=None,
+        **kwargs,
+    ):
+        self._grouping_indices = _grouping_indices or {}
```

Comment:
see above but maybe worthile having a dataclass holding all of these things (so 1 object) e.g.

```
class Groupings:
    indices: .....
```

Reply:
can you do this instead

Reply:
you might also just be able to pass the Grouper object itself.

Reply:
Yeah good point. I'll just pass the grouper object